InTune/InTune/FirebaseServices/DatabaseService.swift

```diff
@@ -10,36 +10,167 @@ import Foundation
+                completion(.success(artists))
+            }
+            
+        }
+    }
+        
+    
+    
+//update fucntion for tags. will create a helper function that saves the "tags" to an array then we will update the database with the array.
+    
+    public func updateUserTags(instruments:[String], genres:[String], completion: @escaping (Result<Bool,Error>) -> () ){
+        guard let user = Auth.auth().currentUser else {return}
+        
+        var tags = [String]()
+        tags.append(contentsOf: instruments)
+        tags.append(contentsOf: genres)
+        
+        db.collection(DatabaseService.artistsCollection).document(user.uid).updateData(["tags":tags]) { (error) in
+            if let error = error {
+                completion(.failure(error))
+            } else {
+                
+                
+                completion(.success(true))
+            }
+        }
+    }
+    
+    public func createVideoPosts(post: ArtistPost, completion: @escaping (Result<Bool, Error>) -> ()) {
+        
+        guard let user = Auth.auth().currentUser else {return}
+        
+        db.collection(DatabaseService.artistsCollection).document(user.uid).collection(DatabaseService.artistPosts).addDocument(data: ["videos" : post.postURL]) { (error) in
+            //add all elements here later
+
+            if let error = error {
+                completion(.failure(error))
+            } else {
+                completion(.success(true))
+            }
+        }
+    }
+    public func createFavoriteArtist(artist: Artist,completion: @escaping (Result<Bool, Error>) -> ()) {
+        guard let user = Auth.auth().currentUser else { return }
+        db.collection(DatabaseService.artistsCollection).document(user.uid).collection(DatabaseService.favCollection).addDocument(data: ["favoritedArtist": artist, "favoritedDate": Timestamp()]) { (error) in
+            if let error = error {
+                completion(.failure(error))
+            } else {
+                completion(.success(true))
+            }
+        }
+    }
+    
+    public func isArtistInFav(for artist: Artist, completion: @escaping (Result<Bool, Error>) -> ()) {
+        guard let user = Auth.auth().currentUser else { return }
+    }
+    
+    //should rename to user
+    public func createGigPost(for user: Artist, gigPost: GigsPost, completion: @escaping (Result<Bool, Error>) ->()) {
+        
+        guard let currentUser = Auth.auth().currentUser else { return }
+        
+        db.collection(DatabaseService.artistsCollection).document(currentUser.uid).collection(DatabaseService.gigPosts).addDocument(data:
+            ["user": user,
+             "gigId": UUID().uuidString,
+             "title": gigPost.title,
+             "descript": gigPost.descript,
+             "photoURL": gigPost.photoURL,
+             "price": gigPost.price,
+             "eventDate": gigPost.eventDate,
+             "createdDate": Timestamp()
```

Comment:
can this whole model be created from wherever we are calling this function and keep this function just call firebase with whatever data it gets?